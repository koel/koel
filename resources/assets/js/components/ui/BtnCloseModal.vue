<template>
  <button data-testid="close-modal-btn" title="Dismiss" type="button" @click.prevent="$emit('click')">
    <icon :icon="faTimes"/>
  </button>
</template>

<script lang="ts" setup>
import { faTimes } from '@fortawesome/free-solid-svg-icons'

defineEmits<{ (e: 'click'): void }>()
</script>

<style lang="scss" scoped>
button {
  display: flex;
  place-content: center;
  place-items: center;
  position: absolute;
  right: 3px;
  top: 3px;
  border-radius: 3px;
  background: var(--color-red);
  width: 22px;
  height: 22px;
  padding: 0;
  color: var(--color-text-primary);
  transition: opacity .3s;

  @media (hover: none) {
    opacity: 1;
  }
}
</style>
