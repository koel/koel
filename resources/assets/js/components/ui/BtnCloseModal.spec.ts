import { expect, it } from 'vitest'
import { screen } from '@testing-library/vue'
import UnitTestCase from '@/__tests__/UnitTestCase'
import BtnCloseModal from './BtnCloseModal.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders', () => expect(this.render(BtnCloseModal).html()).toMatchSnapshot())

    it('emits the event', async () => {
      const { emitted } = this.render(BtnCloseModal)

      await this.user.click(screen.getByRole('button'))

      expect(emitted().click).toBeTruthy()
    })
  }
}
