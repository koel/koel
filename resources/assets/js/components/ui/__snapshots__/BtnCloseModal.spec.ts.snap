// Vitest Snapshot v1

exports[`renders 1`] = `<button data-testid="close-modal-btn" title="Dismiss" type="button" data-v-2d6954b5=""><br data-testid="icon" icon="[object Object]" data-v-2d6954b5=""></button>`;
